### PR TITLE
Python UAT: Alice sends tokens to bob.aet

### PR DESCRIPTION
**Pull request by velzevur**
[Pivotal story](https://www.pivotaltracker.com/story/show/154277512)
UAT test to verify that Bob can register a name, using this name Alice can resolve Bob's address and send tokens to that address.

Test steps:
1. Setup
1.1 Generate pub-priv pair and public address for Alice and Bob
1.2 Node is started, miner mines some blocks and sends tokens to Alice and Bob (so both of them have accounts in the chain and tokens to spend)
2. Bob registers the name `bob.aet`
2.1 Using a `salt` and the name, Bob prepares a `commitement_hash`.
2.2 Bob prepares a `aens_preclaim_tx`, signes it and posts it to the node. Waits a couple of blocks so it is included. Bob has a `name_hash`
2.3 Using the `salt` from 2.1 and the `name_hash` of 2.2, Bob prepares a `aens_claim_tx`, signes it and posts it to the node. Waits a couple of blocks so it is included. Bob now has registered the name but it still doesn't have any address related to it.
2.4 Bob prepares `aens_update_tx` in which he sets his address in the `pointers`, signes it and posts it to the node. Waits a couple of blocks so it is included.
2.5 Bob validates his name is in the nameservice
3. Alice sends to bob.aet
3.1 Alice resolves the name and gets Bob's address from the `pointers` attached to that name
3.2 Alice sends tokens to that address
4. Validation
4.1 Alice's balance is updated
4.1 Bob's balance is updated

### CODE REVIEW

File: py/tests/integration/test_spend_tx.py [hunk 1]
```diff
@@ -6,14 +6,21 @@
...
+from swagger_client.models.name_claim_tx import NameClaimTx
+from swagger_client.models.name_update_tx import NameUpdateTx
 from swagger_client.rest import ApiException
 
+import keys
+
 settings = common.test_settings(__name__.split(".")[-1])
 
-def test_successful():
+def _test_successful():
```
Comment (lucafavatella) [inline py/tests/integration/test_spend_tx.py]: What is impact of prefixing _ to name? Why?

File: py/tests/integration/test_spend_tx.py [hunk 2]
```diff
@@ -80,7 +87,7 @@ def test_successful():
     common.stop_node(alice_node)
     shutil.rmtree(root_dir)
 
-def test_not_enough_tokens():
+def _test_not_enough_tokens():
```
Comment (lucafavatella) [inline py/tests/integration/test_spend_tx.py]: What is impact of prefixing _ to name? Why?

File: py/tests/integration/test_spend_tx.py (hunk 1 above)
Comment (velzevur) [reply py/tests/integration/test_spend_tx.py]: Ah, typo

File: py/tests/integration/test_unsigned_tx.py [hunk 3]
```diff
@@ -239,7 +239,7 @@ def send_tokens_to_user(user, test_settings, internal_api, external_api):
     spend_tx_obj = SpendTx(
         recipient_pubkey=test_settings[user]["pubkey"],
         amount=test_settings[user]["amount"],
-        fee=test_settings[user]["amount"])
+        fee=test_settings[user]["fee"])
```
Comment (lucafavatella) [inline py/tests/integration/test_unsigned_tx.py]: Well spotted.

File: py/tests/integration/test_spend_tx.py [hunk 4]
```diff
@@ -175,4 +240,70 @@ def setup_node_with_tokens(test_settings, node_name):
...
+
+    top = external_api.get_top()
+    common.wait_until_height(external_api, top.height + 3)
+
+def register_name(name, address, external_api, private_key):
+    salt = 42
+    commitment = external_api.get_commitment_hash(name, salt).commitment
+
+    # preclaim
+    unsinged_preclaim = common.base58_decode(\
```
Comment (lucafavatella) [inline py/tests/integration/test_spend_tx.py]: unsinged_preclaim -> unsigned_preclaim

File: py/tests/integration/test_spend_tx.py [hunk 5]
```diff
@@ -175,4 +240,70 @@ def setup_node_with_tokens(test_settings, node_name):
...
+            NamePreclaimTx(commitment=commitment, fee=1, account=address)).tx)
+    signed_preclaim = keys.sign_encode_tx(unsinged_preclaim, private_key)
+
+    external_api.post_tx(Tx(tx=signed_preclaim))
+    top = external_api.get_top()
+    common.wait_until_height(external_api, top.height + 3)
+
+    # claim
+    encoded_name = common.encode_name(name)
+    unsinged_claim = common.base58_decode(\
```
Comment (lucafavatella) [inline py/tests/integration/test_spend_tx.py]: unsinged_claim -> unsigned_claim

File: py/tests/integration/test_spend_tx.py [hunk 6]
```diff
@@ -143,6 +150,64 @@ def test_not_enough_tokens():
...
+    send_tokens_to_name(bob_name, tokens_to_send, alice_address, alice_private_key, ext_api)
+
+    # validate balances
+    alice_balance2 = common.get_account_balance(int_api, pub_key=alice_address).balance
+    bob_balance2 = common.get_account_balance(int_api, pub_key=bob_address).balance
+
+    print("Alice balance is " + str(alice_balance2))
+    print("Bob balance is " + str(bob_balance2))
+
+    assert_equals(alice_balance2, alice_balance0 - tokens_to_send - 1)
```
Comment (lucafavatella) [inline py/tests/integration/test_spend_tx.py]: I understand this 1 is meant to be test_settings["spend_tx"]["fee"] (had it been defined).

Comment (velzevur) [reply py/tests/integration/test_spend_tx.py]: Yes, I though I'd get a cleaner interface if remove the fee from the settting

**Pull request merged**